library/Opus/Enrichment.php

```diff
@@ -65,7 +66,10 @@ class Opus_Enrichment extends Opus_Model_Dependent_Abstract {
      */
     protected function _init() {
         $key = new Opus_Model_Field('KeyName');
-        $key->setMandatory(true)->setValidator(new Zend_Validate_NotEmpty());
+        $key->setMandatory(true)
+                ->setValidator(new Zend_Validate_NotEmpty())
+                ->setSelection(true)
```

Comment:
Ich hatte diese und die nächste Zeile im Rahmen von OPUSVIER-2494 entfernt. Ausversehen wieder dazugekommen oder Absicht? Set/getSelection und get/setDefault wurden zu dem Zeitpunkt nicht mehr verwendet. Diese Funktion stehen hauptsächlich im Zusammenhang mit der Präsentation und die will ich nicht mehr im Datenmodell haben. Ein getAll für ein Formularfeld ist in Ordnung, aber ein getAll für jedes einzelne `Opus_Enrichment` Objekt nicht.

Reply:
Deine Änderung durch OPUSVIER-2494 habe ich nicht mitbekommen, sorry. Wenn trotz des Entfernens der beiden Methodenaufrufe (`setSelection` sowie `setDefault`) das Admin-Formular für das Editieren von Dokumenten noch korrekt funktioniert, sehe ich keine Probleme. An den Enrichments habe ich nur eine Kleinigkeit geändert: die beiden Spezial-Enrichments für DOIs und URNs werden nicht als solche angezeigt, sondern sind nur über die entsprechenden Checkboxen editierbar.

Hast Du vor die beiden Methoden gleich in der Basis-Klasse `Opus_Model_Field` zu entfernen?

Reply:
Momentan taste ich mich für OPUS 4.7 ganz vorsichtig an die Änderungen der Modell-Klassen ran. Die APIs bleiben erst einmal unverändert. Die Ideen hinter diesen Funktion sind ja auch nicht schlecht. Leider sind sie nirgendwo dokumentiert und in der Praxis hat uns die Integration von Datenmodell und Präsentation (`setSelection` etc.) Probleme bereitet. Im aktuellen Code haben wir mehr Redundanzen, was das Datenmodell angeht, aber die Umsetzung eines Form-Builders, der den Anpassungswünschen gerecht wird, war zu aufwendig und komplex. Ich wäre nicht überrascht, wenn wir uns später langsam wieder in Richtung automatisch generierter Formulare bewegen. Für das Publish-Model brauchen wir das sowieso.